server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -826,7 +875,7 @@ public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep) {
     public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep, ActionListener<Void> listener) {
         if (canDeleteStaleCommits.compareAndSet(true, false)) {
             try {
-                threadPool.executor(ThreadPool.Names.REMOTE_PURGE).execute(() -> {
+                threadPool.executor(getExecutorNameForAsyncOperations()).execute(() -> {
```

Comment:
Can we use this instead 

```
                String executorName = ThreadPool.Names.REMOTE_PURGE;
                if (Thread.currentThread().getName() == executorName) {
                    executorName = ThreadPool.Names.SAME;
                }
                    
                threadPool.executor(executorName).execute(() -> {
```

Reply:
rolled back this logic, it is not needed anymore.